File: MatchModule_New/Games.NB_MatchModule.BLL/Model/Creatures/Player.IStopball.cs
/********************************************************************************
 * 文件名：PlayerStopball
 * 创建人：
 * 创建时间：2009-12-18 22:38:02
 * 版本：1.0
 * 本文件版本号：1.0
 * 最后更新：
 * 功能说明：This partial class implemented the interface of the <see cref="IStopball"/>.
 * 历史修改记录：
 * <author>  <time>           <version >   <desc>
 *********************************************************************************/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Games.NB.Match.Base.Interface.Player;
using Games.NB.Match.Log;
using Games.NB.Match.Base;

namespace Games.NB.Match.BLL.Model.Creatures {

    /// <summary>
    /// Represents the player entity.
    /// This partial class implemented the interface of the <see cref="IStopball"/>.
    /// </summary>
    public sealed partial class Player {
        #region IStopball Members

        public void Stopball()
        {
            if (this._status.Hasball == false || this.Status.BallDistance > Defines.Player.DEFENCE_RANGE)
            {
                throw new NotSupportedException("[停球]球在不在这个球员身上：" + this);
            }

            LogHelper.Insert(String.Format("[停球]{0} 停球", this));

            this.Manager.Match.Football.Kick(this.Manager.Match.Football.Current, 0, this);
        }

        #endregion
    }
}

File: SVN球场AI/HB_SNS/Player/Player_IStopball.cpp
/********************************************************************************
 * 文件名：Player_IStopball.cpp
 * 创建人：宋斌
 * 创建时间：2011-6-10
 * 版本：1.0
 * 本文件版本号：1.0
 * 最后更新：
 * 功能说明：
 * 历史修改记录：
 * <author>  <time>           <version >   <desc>
 *********************************************************************************/
#include "Player.h"

void Player::Stopball() 
{
    if (m_Status->Hasball() == false || m_Status->BallDistance() > m_Attribute->DefenceRange())
    {
        throw NotSupportedException("[停球]球在不在这个球员身上：" + lexical_cast<string>(m_Id));
    }

    LogHelper::Insert("[停球]" + lexical_cast<string>(m_Id) + "停球");

    m_Manager->GetMatch()->GetFootball()->Kick(m_Manager->GetMatch()->GetFootball()->Current(), 0, this);
}
